src/main/java/http_server/HTTPSession.java

```diff
@@ -0,0 +1,123 @@
+package http_server;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class HTTPSession implements Runnable {
+    private final SocketWrapper client;
+    private Boolean isRunning;
+    private String crlf = "\r\n";
+
+    public HTTPSession(SocketWrapper client) {
+        this.client = client;
+        this.isRunning = true;
+    }
+
+    public void run() {
+        while (!Thread.interrupted() && isRunning) {
+            if (!client.ready()) {
+                try {
+                    Thread.sleep(1000);
+                    continue;
+                } catch (InterruptedException e) {
+                    close();
+                    return;
+                }
+            }
+
+            String newLine = client.readData();
+
+            System.err.println(newLine);
+
+            String[] request = newLine.trim().split(crlf);
```

Comment:
Should this class know so much about how a request is constructed (the spacing/components of a request, the body, etc)?

Reply:
this is in the middle of a refactor! :D